src/persist/src/operators/stream.rs

```diff
@@ -41,46 +50,215 @@ where
 {
     fn persist(
         &self,
-        token: (StreamWriteHandle<K, V>, StreamReadHandle<K, V>),
+        name: &str,
+        write: StreamWriteHandle<K, V>,
+    ) -> (
+        Stream<G, ((K, V), u64, isize)>,
+        Stream<G, (String, u64, isize)>,
+    ) {
+        let operator_name = format!("persist({})", name);
+        let mut persist_op = OperatorBuilder::new(operator_name.clone(), self.scope());
+
+        let mut input = persist_op.new_input(&self, Pipeline);
+
+        let (mut data_output, data_output_stream) = persist_op.new_output();
+        let (mut error_output, error_output_stream) = persist_op.new_output();
+
+        let mut buffer = Vec::new();
+        let mut write_futures = HashMap::new();
+        let mut input_frontier =
+            Antichain::from_elem(<G::Timestamp as timely::progress::Timestamp>::minimum());
+
+        persist_op.build(move |_capabilities| {
+            move |frontiers| {
+                let mut data_output = data_output.activate();
+                let mut error_output = error_output.activate();
+
+                // write out everything and forward, keeping the write futures
```

Comment:
nit: the persist crate tends to use capitalization and punctuation in comments. here and elsewhere